build_wheels.py

```diff
@@ -0,0 +1,227 @@
+        for feature in idf_requirements_json['features']:
+            res = requests.get(f"{idf_resources_url}{branch}/{feature['requirement_path']}")
+            if check_response(res, f"Failed to download feature (requirement group) '{feature['name']}'"):
+                requirements_txt = res.text.splitlines()
+                print(f"Added ESP-IDF {feature['name']} requirements")
+        return requirements_txt
+
+    def download_branch_constraints() -> list:
+        """Download constraints file for specific branch
+
+        Returns:
+            list: lines from constraints.vX.Y.txt
+        """
+        res = requests.get(constraint_file_url)
+        if check_response(res, f'Failed to download ESP-IDF constraints file {idf_constraints[i]} for branch {branch}'):
+            requirements_txt = res.text.splitlines()
+            print(f'Added ESP-IDF constraints file {idf_constraints[i]} for branch {branch}')
+        return requirements_txt
+
+    def add_into_requirements() -> set:
+        """Create set of requirements from downloaded lines of requirements
+            - set is used to prevent duplicates
+
+        Returns:
+            set: IDF requirements
+        """
+        requirements_set = set()
+        for line in map(str.strip, requirements_txt):
+            # check if in the line or the line itself is not a comment
+            line = line.split('#', 1)[0]
+            if line:
+                try:
+                    requirements_set.add(Requirement(line))
+                except InvalidRequirement:
+                    # TODO not a classic requirement (e.g. '--only-binary cryptography')
+                    requirements_set.add(line)
+        return requirements_set
+
+    for i, branch in enumerate(idf_branches):
+        idf_requirements_json_url = f'{idf_resources_url}{branch}/tools/requirements.json'
+        constraint_file_url = f'https://dl.espressif.com/dl/esp-idf/espidf.constraints.{idf_constraints[i]}.txt'
+
+        res = requests.get(idf_requirements_json_url)
+        if check_response(res, f'\nFailed to download requirements JSON for branch {branch}'):
+            idf_requirements_json = json.loads(res.content)
+        else:
+            continue
+
+        requirements_txt += download_branch_requirements()
+        requirements_txt += download_branch_constraints()
+
+    return add_into_requirements()
+
+
+##############################################################
+def print_requirements(requirements_set: set):
+    """Prints assembled list of requirements
+
+    Args:
+        requirements_set (set): IDF requirements
+    """
+    print('\n---------- REQUIREMENTS ----------')
+    for req in requirements_set:
+        print(req)
+
+
+##############################################################
+def main():
+    """Builds Python wheels for IDF dependencies
+    - Automatically fetch IDF branches and corresponding constraints files
```

Comment:
These details in doc-string seem unnecessary, because the same is obvious from the names of the called functions (you have them well named).